File: Fines2Wpf/Data/horario.cs
#nullable enable
using SqlOrganize;
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Reflection;
using Utils;

namespace Fines2Wpf.Data
{
    public class Data_horario : SqlOrganize.Data
    {

        public Data_horario ()
        {
            Initialize();
        }

        public Data_horario(DataInitMode mode)
        {
            Initialize(mode);
        }

        protected virtual void Initialize(DataInitMode mode = DataInitMode.Null)
        {
            switch(mode)
            {
                case DataInitMode.Default:
                case DataInitMode.DefaultMain:
                    _id = (string?)ContainerApp.db.Values("horario").Default("id").Get("id");
                break;
            }
        }

        public string? Label { get; set; }

        protected string? _id = null;
        public string? id
        {
            get { return _id; }
            set { _id = value; NotifyPropertyChanged(); }
        }
        protected DateTime? _hora_inicio = null;
        public DateTime? hora_inicio
        {
            get { return _hora_inicio; }
            set { _hora_inicio = value; NotifyPropertyChanged(); }
        }
        protected DateTime? _hora_fin = null;
        public DateTime? hora_fin
        {
            get { return _hora_fin; }
            set { _hora_fin = value; NotifyPropertyChanged(); }
        }
        protected string? _curso = null;
        public string? curso
        {
            get { return _curso; }
            set { _curso = value; NotifyPropertyChanged(); }
        }
        protected string? _dia = null;
        public string? dia
        {
            get { return _dia; }
            set { _dia = value; NotifyPropertyChanged(); }
        }
        protected override string ValidateField(string columnName)
        {

            switch (columnName)
            {

                case "id":
                    if (_id == null)
                        return "Debe completar valor.";
                    return "";

                case "hora_inicio":
                    if (_hora_inicio == null)
                        return "Debe completar valor.";
                    return "";

                case "hora_fin":
                    if (_hora_fin == null)
                        return "Debe completar valor.";
                    return "";

                case "curso":
                    if (_curso == null)
                        return "Debe completar valor.";
                    return "";

                case "dia":
                    if (_dia == null)
                        return "Debe completar valor.";
                    return "";

            }

            return "";
        }
    }
}

File: Data/Horario.php
<?php

namespace SqlOrganize\Sql\Fines2;

use SqlOrganize\Sql\Entity;
use SqlOrganize\Sql\Db;
use Exception;
use DateTime;

class Horario extends Entity
{

    public function __construct(Db $db)
    {
        $this->_entityName = "horario";
        $this->_db = $db;
        $this->setDefault();
    }

    public function setFromTree(array $treeData)
    {
    $curso_ = null;
    $dia_ = null;
    }

    /** @var string|null */
    public ?string $curso = null;

    /** @var string|null */
    public ?string $dia = null;

    /** @var DateTime|null */
    public ?DateTime $hora_fin = null;

    /** @var DateTime|null */
    public ?DateTime $hora_inicio = null;

    /** @var string|null */
    public ?string $id = null;

    /** @var Curso|null (fk horario.curso _m:o curso.id) */
    public ?Curso $curso_ = null;

    /** @var string|null */
    public ?string $curso__ = null;

    /** @var Dia|null (fk horario.dia _m:o dia.id) */
    public ?Dia $dia_ = null;

    /** @var string|null */
    public ?string $dia__ = null;

}
